packages/salesforcedx-vscode-apex/src/oas/generationStrategy/generationStrategyFactory.ts

```diff
@@ -26,8 +26,13 @@ export const initializeAndBid = async (
 }> => {
   // Initialize strategies
   const strategies = new Map<GenerationStrategyType, StrategyTypes>();
-  strategies.set('ApexRest', new ApexRestStrategy(metadata, context));
-  strategies.set('AuraEnabled', new AuraEnabledStrategy(metadata, context));
+  const apexRestStrategy = new ApexRestStrategy(metadata, context);
+  await apexRestStrategy.initialize();
```

Comment:
the thing I don't like about this pattern is that after `new` you have to initialize before using it or bad things will happen.  

I think I'd rather it have a `private` constructor (to prevent `new`) and have `initialize` call that, and then return an initialized class instance.  Safer to prevent developer errors than expect them to know there's multiple steps required to get a usable instance.